src/mame/brother/lw30.cpp

```diff
@@ -0,0 +1,1105 @@
+				floppy_steps++;
+			else
+				LOGFLOPPY("%s: illegal step %02x=>%02x\n", machine().describe_context(), floppy_control, data);
+			break;
+		default:
+			LOGFLOPPY("%s: initial step %02x=>%02x\n", machine().describe_context(), floppy_control, data);
+			break;
+		}
+		auto new_track = floppy_steps / 4;
+		auto floppy_device = floppy->get_device();
+		if(new_track != old_track) {
+			floppy_device->dir_w(new_track < old_track);
+			floppy_device->stp_w(true);
+			floppy_device->stp_w(false);
+		}
+		LOGFLOPPY("%s: floppy_steps=%3d => old_track=%2d new_track=%2d cyl=%2d\n", machine().describe_context(), floppy_steps, old_track, new_track, floppy_device->get_cyl());
+		assert(floppy_device->get_cyl() == new_track);
+		floppy_control = (floppy_control & 0xf0) | (data & 0x0f);
+	}
+
+	uint8_t io_98_r() {
+		// mirrored in RAM
+		// bit 0: cleared in rst28_06 in mirror
+		// bit 2: cleared before formatting in mirror; set after formatting
+		// bit 3: cleared before formatting in mirror
+		// bit 4: cleared before writing in mirror; set after writing
+		if(io_88 & 0b10)
+			floppy_read_until_zerobit = true;
+		else
+			floppy_read_until_zerobit = false;
+
+		LOGFLOPPY("%s: read %02X from IO 98\n", machine().describe_context(), io_98);
+		return io_98;
+	}
+	void io_98_w(uint8_t data) {
+		LOGFLOPPY("%s: write %02X to IO 98\n", machine().describe_context(), data);
+		io_98 = data;
+	}
+
+	uint8_t illegal_io_r(offs_t offset, uint8_t mem_mask = ~0) {
+		LOGFLOPPY("%s: unmapped IO read from %0*X & %0*X\n", machine().describe_context(), 4, offset + 0x40, 2, mem_mask);
+		return 0;
+	}
+	void illegal_io_w(offs_t offset, uint8_t data, uint8_t mem_mask = ~0) {
+		LOGFLOPPY("%s: unmapped IO write to %0*X = %0*X & %0*X\n", machine().describe_context(), 4, offset + 0x40, 2, data, 2, mem_mask);
+	}
+
+	uint8_t io_b0_r() {
+		// Tetris reads bit 3, needed for correct keyboard layout
+		return 0b1000;
+	}
+	uint8_t io_b8_r() { // B8 (keyboard)
+		// keyboard matrix
+		if(io_b8 <= 8)
+			return m_io_kbrow[io_b8].read_safe(0);
+		return 0x00;
+	}
+	void io_b8_w(uint8_t data) {
+		io_b8 = data;
+	}
+
+	// Tetris Game Over Melody:
+	// according to oscilloscope:
+	// - total length 2.630s, wavosaur shows 2.043s
+	// - last note: 600µs on, 600µs off, on-to-on: 1260µs, wavosaur shows 1083µs
+	// according to MAME
+	// - 11µs between writes to beeper
+	// - 476µs (value=0x310=784) => 0.6071x between writes of 01 and 81 to beeper
+	// - 713µs
+	// - 724µs
+	// - last note: 543µs (value=0x2ba=698) => 0.7779x
+
+	void beeper_w(uint8_t data) { // F0
+		#if 0
+			static uint8_t lastData{};
+			static attotime last{};
+			static attotime lastDifferent{};
+			attotime cur{ maincpu->local_time() };
+
+			LOG("beeper_w(%02X) @ %s (delta=%sµs diff_delta=%sµs)\n", data, cur.as_string(), ((cur - last) * 1000000).as_string(0), ((cur - lastDifferent) * 1000000).as_string(0));
+			if(data != lastData)
+				lastDifferent = cur;
+			last = cur;
+			lastData = data;
+		#endif
```

Comment:
Please use `if (false)` or something to disable code so it will still be seen by the compiler and has less change of rotting.  Also, please add a comment for why it’s disabled.